pandas/tests/frame/methods/test_astype.py

```diff
@@ -767,13 +764,9 @@ def test_astype_dt64_to_string(
         if frame_or_series is DataFrame:
             item = item.iloc[0]
         if using_infer_string:
-            assert item is np.nan
-        else:
             assert item is pd.NA
-
-        # For non-NA values, we should match what we get for non-EA str
-        alt = obj.astype(str)
-        assert np.all(alt.iloc[1:] == result.iloc[1:])
+        else:
+            assert item is np.nan
```

Comment:
item should never be np.nan with the original `string` dtype?

Reply:
Nice catch - I'll take a closer look as to why that happens

Reply:
there should be no need for inference here. the result and expected are both astyped. I would expect that the using_infer_string fixture is not needed at all. @jorisvandenbossche has asked that you also test with `astype("str")` and that would not change any inference. There is a fixture for testing the different the string dtypes.

Reply:
It appears that the error message `NotImplementedError: eq not implemented for <class 'pandas.core.arrays.string_.StringArray'>` is misleading.

if you do `result = obj.astype("string[pyarrow]")` this test passes and only fails for the numpy backed object array. i.e. the default for obj.astype("string") is obj.astype("string[python]")

In the `<ArrowStringArrayNumpySemantics>` code...

```python
    def _cmp_method(self, other, op) -> ArrowExtensionArray:
        pc_func = ARROW_CMP_FUNCS[op.__name__]
        if isinstance(
            other, (ArrowExtensionArray, np.ndarray, list, BaseMaskedArray)
        ) or isinstance(getattr(other, "dtype", None), CategoricalDtype):
            try:
                result = pc_func(self._pa_array, self._box_pa(other))
            except pa.ArrowNotImplementedError:
```

for the object backed _string_ array, _other_ is not an instance of `ArrowExtensionArray` but is an instance of `StringArray` and although the comparison is skipped and a _NotImplementedError_ raised, it appears that `pc_func(self._pa_array, self._box_pa(other))` does give the expected result with a numpy backed object `StringArray`

Reply:
> it appears that `pc_func(self._pa_array, self._box_pa(other))` does give the expected result with a numpy backed object `StringArray`

Yes, that is also what I would expect